### PR TITLE
[macOS] NSGestureRecognizer.State should be get/set like UIKit's

### PR DESCRIPTION
**Pull request by chamons**
- https://bugzilla.xamarin.com/show_bug.cgi?id=60935
- This allows subclassing to work

### CODE REVIEW

File: src/appkit.cs [hunk 1]
```diff
@@ -7457,7 +7457,7 @@ interface NSGestureRecognizer : NSCoding {
 		Selector Action { get; set; }
 
 		[Export ("state")]
-		NSGestureRecognizerState State { get; }
+		NSGestureRecognizerState State { get; set; }
```
Comment (timrisi) [inline src/appkit.cs]: Is it normally settable from obj-c?  If not, should there be an [Advice] or anything on the base version?

Comment (chamons) [reply src/appkit.cs]: From the bug:

```
So in obj-c they declare it read only

`@property (readonly) NSGestureRecognizerState state;`

and later use this to all subclases to set
    
    @interface NSGestureRecognizer (NSSubclassUse)
    
    // readonly for users of a gesture recognizer. may only be changed by direct subclasses of NSGestureRecognizer
    @property NSGestureRecognizerState state;  // the current state of the gesture recognizer. can only be set by subclasses of NSGestureRecognizer, but can be read by consumers
    
    
In iOS we handle this correctly:

		[Export ("state")]
		UIGestureRecognizerState State { get; set; }

but do not in XM:

		[Export ("state")]
		NSGestureRecognizerState State { get; }
```

Comment (chamons) [reply src/appkit.cs]: I should have added some of it to the commit message. I will on squash.

Comment (timrisi) [reply src/appkit.cs]: Yah, I got what it was doing from the bug, I was more wondering if the setter in the base class should have something like [Advice ("Only for use in subclasses")] or something like that.  Not sure if you'd ever being using a base NSGestureRecognizer to see the advice, but figured its worth asking.

Comment (chamons) [reply src/appkit.cs]: Not a terrible idea.

**Comment by chamons** [comment]
Bah. Typo due to lack of . in advise.

File: src/appkit.cs [hunk 2]
```diff
@@ -7457,7 +7457,7 @@ interface NSGestureRecognizer : NSCoding {
 		Selector Action { get; set; }
 
 		[Export ("state")]
-		NSGestureRecognizerState State { get; }
+		NSGestureRecognizerState State { get; [Advice ("For Subclass Use.")] set; }
```
Comment (VincentDondain) [inline src/appkit.cs]: I'm not a huge fan of the wording of the advice, I find it a little unclear and I had to open the bug to be sure.

I'd go for something more verbose: `Only subclasses of 'NSGestureRecognizer' can set this property.`

Also there's no reason to have capital letter for "Subclass" and "Use" (at least let's fix that).

**Pull request merged**